src/util/IOUtil.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @ignore *//** */
+import {
+    BigDecimal,
+    LocalDate,
+    LocalDateTime,
+    LocalTime,
+    OffsetDateTime,
+} from '../core';
+import {bufferToBigInt, bigIntToBuffer} from './BigDecimalUtil';
+import {DataInput, DataOutput} from '../serialization';
+
+/** @internal */
+export class IOUtil {
+
+    static readDecimal(inp: DataInput): BigDecimal {
+        const buffer = inp.readByteArray();
+        const scale = inp.readInt();
+        return new BigDecimal(bufferToBigInt(buffer), scale);
+    }
+
+    static readLocalTime(inp: DataInput): LocalTime {
+        const hour = inp.readByte();
+        const minute = inp.readByte();
+        const second = inp.readByte();
+        const nano = inp.readInt();
+        return new LocalTime(hour, minute, second, nano);
+    }
+
+    static readLocalDate(inp: DataInput): LocalDate {
+        const year = inp.readInt();
+        const month = inp.readByte();
+        const date = inp.readByte();
+        return new LocalDate(year, month, date);
+    }
+
+    static readLocalDatetime(inp: DataInput): LocalDateTime {
```

Comment:
```suggestion
    static readLocalDateTime(inp: DataInput): LocalDateTime {
```

Reply:
FYI: I did a mass renaming following this